lib/puppet/provider/package/apt.rb

```diff
@@ -44,12 +48,45 @@ def checkforcdrom
     end
   end
 
+  def best_version(should_range)
+    available_versions = SortedSet.new
+
+    output = aptcache :madison, @resource[:name]
+    output.each_line do |line|
+      is = line.split('|')[1].strip
+      begin
+        is_version = DebianVersion.parse(is)
+        available_versions << is_version
+      rescue DebianVersion::ValidationFailure
+        Puppet.warning("Cannot parse #{is} as a debian version")
+      end
+    end
+
+    included_available_version = available_versions.select do |version|
+      should_range.include?(version)
+    end
+
```

Comment:
if `available_versions` is a sorted set do we need to select all the valid ones? can't we just use `find` which will return the first occurence?